cmd/soroban-cli/src/commands/contract/info/env_meta.rs

```diff
@@ -0,0 +1,72 @@
+use std::fmt::Debug;
+
+use clap::{command, Parser};
+use stellar_xdr::curr::ScEnvMetaEntry;
+
+use soroban_spec_tools::contract;
+use soroban_spec_tools::contract::Spec;
+
+use crate::commands::contract::info::env_meta::Error::{NoEnvMetaPresent, NoSACEnvMeta};
+use crate::commands::contract::info::shared;
+use crate::commands::contract::info::shared::{fetch_wasm, MetasInfoOutput};
+
+#[derive(Parser, Debug, Clone)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub common: shared::Args,
+    /// Format of the output
+    #[arg(long, default_value = "text")]
+    pub output: MetasInfoOutput,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Wasm(#[from] shared::Error),
+    #[error(transparent)]
+    Spec(#[from] contract::Error),
+    #[error("Stellar asset contract doesn't contain meta information")]
+    NoSACEnvMeta(),
+    #[error("no meta present in provided WASM file")]
+    NoEnvMetaPresent(),
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<String, Error> {
+        let bytes = fetch_wasm(&self.common).await?;
+
+        if bytes.is_none() {
+            return Err(NoSACEnvMeta());
+        }
+        let spec = Spec::new(&bytes.unwrap())?;
+
+        if spec.env_meta_base64.is_none() {
+            return Err(NoEnvMetaPresent());
+        }
+
+        let res = match self.output {
+            MetasInfoOutput::XdrBase64 => spec.env_meta_base64.unwrap(),
```

Comment:
Where possible it is best in Rust to avoid the use of .unwrap and instead safely unwrap using pattern matching.

Using is_none/is_some and unwrap means there will be two separate lines of code that must be logically synchronized, and if they ever get out of sync the unwrap will panic.

For example:
```rust
let Some(bytes) = bytes else {
    return Err(NoSACEnvMeta());
};
```

Reply:
Good point. I was trying to reduce amount of nested layers

Reply:
That resonates with me. It's sort of something I found I had to let go of coming into Rust. I was used to writing Go / Java / Ruby that favors early exit patterns rather than nested if..else, but in Rust the latter just works better with unwrapping and pattern matching most of the time. 🤷🏻